swaylock/main.c

```diff
@@ -115,14 +115,100 @@ void notify_key(enum wl_keyboard_key_state state, xkb_keysym_t sym, uint32_t cod
 	}
 }
 
+static void cairo_set_source_u32(cairo_t *cairo, uint32_t color) {
```

Comment:
I had not considered that issue, so I'm actually not sure where to put it now. I bet people will complain if one thing becomes a hard dependency, so it should be put somewhere neutral.